### DISCUSSION TITLE
Has any plan Importing  ` vite-plugin-components`

### DISCUSSION THREAD

**Discussion by fyeeme** [quasarframework/quasar#10013] [General - Components / Directives / etc]
**Is your feature request related to a problem? Please describe.**
Yes support vite

**Describe the solution you'd like**
 quasar base vite auto import

**Describe alternatives you've considered**
no

**Additional context**
we can use quasar on vite ,but  if support [vite-plugin-components](https://github.com/antfu/vite-plugin-components), it would be better.

I know quasar has its plan to support on Q4, but if we can support  vite-plugin-components, it would be much better. 

also feel free to try [vite-quasar ](https://github.com/fyeeme/vite-quasar) out of box

**Comment by AGPDev** [comment]
i use this to resolve componentes with vite-plugin-components

`resolver.js`
```js
import { existsSync } from 'fs'
import { resolve } from 'path'
import importMap from 'quasar/dist/transforms/import-map.json'

export default name => {
  const importName = importMap[name]
  if (!importName) {
    return
  }

  const script = `quasar/${importName}`
  let style = script.replace(/\.js$/, '.sass')

  if (!existsSync(resolve('node_modules', style))) {
    style = undefined
  }

  return {
    path: script,
    sideEffects: style,
  }
}
```

`vite.config`
```js
ViteComponents({
    customComponentResolvers: [
    	resolveQuasar,
    ]
})
```

**Reply by fyeeme** [reply]
@AGPDev 
Many thanks for your replay, seem a lit trouble to use this.  main problem is can not fond `qusar.sass` ，even i mannly import it, still can not found.
![image](https://user-images.githubusercontent.com/6078758/125035257-28f09780-e0c4-11eb-821e-ea870eca5cfa.png)


main.js
```js
import { createApp } from 'vue'
import App from './App.vue'

// import { Quasar } from 'quasar'
// import { quasarConfig } from 'src/plugins/quasar.js'

import router from './router'
import store from './store'
// createApp(App).use(Quasar, quasarConfig).use(router).use(store).mount('#app')

createApp(App).use(router).use(store).mount('#app')

```

vite.config.js

```js
import { defineConfig } from 'vite'
import { resolve } from 'path'

import vue from '@vitejs/plugin-vue'
import Pages from 'vite-plugin-pages'
import Layouts from 'vite-plugin-vue-layouts'
import ViteComponents from 'vite-plugin-components'
import resolveQuasar from './src/plugins/resolver.js'
import ViteIcons, { ViteIconsResolver } from 'vite-plugin-icons'
// https://vitejs.dev/config/
export default defineConfig({
  plugins: [
    vue(),
    Pages(),
    Layouts(),
    ViteIcons(),
    ViteComponents({
      customComponentResolvers: [
        ViteIconsResolver({
          componentPrefix: 'icon',
        }),
        resolveQuasar,
      ],
    }),
  ],

})

```
and if convenience, this is [my repo](https://github.com/fyeeme/vite-quasar/tree/feature/vite-plugin-component) , and help me see what's going on. 

thanks again. bro!.
